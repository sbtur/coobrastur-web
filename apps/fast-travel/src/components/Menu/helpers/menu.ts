import { MenuItem } from '../types/Menu.types';

export const MENU_ITEMS: MenuItem[] = [
  {
    label: 'O Clube',
    href: '#',
    items: [
      {
        label: 'O Grupo Coobrastur',
        href: '#',
        description: 'Lorem ipsum dolor sit amet, consectetur.',
      },
      {
        label: 'Como Funciona',
        href: '#',
        description: 'Lorem ipsum dolor sit amet, consectetur.',
      },
      {
        label: 'Dúvidas Frequentes',
        href: '#',
        description: 'Lorem ipsum dolor sit amet, consectetur.',
      },
    ],
  },
  {
    label: 'Planos',
    href: '/plans',
  },
  {
    label: 'Agência de Viagens',
    href: '/travel-agency',
  },
  {
    label: 'Hotéis',
    href: '/search-place',
  },
];
