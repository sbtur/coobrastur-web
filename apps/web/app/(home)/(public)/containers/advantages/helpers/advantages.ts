import { Advantage } from '../types/advantage.types';

export const ADVANTAGES: Advantage[] = [
  {
    icon: '/images/icons/hotel.svg',
    title: 'Mais de 2.000 hotéis',
    description:
      'Hotéis selecionados a dedo para você viver experiências incríveis, com economia e café da manhã sempre garantido para você e um acompanhante.',
  },
  {
    icon: '/images/icons/wallet-money.svg',
    title: 'Pagamento Flexível',
    description:
      '12 mensalidades ao ano com um valor que cabe no seu bolso e com pagamento que não ocupa o limite do seu cartão de crédito.',
  },
  {
    icon: '/images/icons/money-plus.svg',
    title: 'Economia certa',
    description:
      'Com nossa Assinatura, você garanta até 60% de economia no valor das diárias em hotéis em comparação a sites de hospedagem.',
  },
  {
    icon: '/images/icons/chat-square-call.svg',
    title: 'Atendimento personalizado',
    description:
      'Ajudamos você a viver viagens inesquecíveis com suporte especializado em todas as etapas, do planejamento ao check-out.',
  },
];
