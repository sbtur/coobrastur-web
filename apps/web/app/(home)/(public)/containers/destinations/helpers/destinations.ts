import { DestinationResponse } from '../types/destination.types';

export const DESTINATIONS: DestinationResponse[] = [
  {
    id: '1',
    image: '/images/mocks/places/place-1.jpg',
    city: {
      slug: 'gramado',
      name: 'Gramado',
    },
    country: {
      slug: 'brasil',
      name: 'Brasil',
    },
    state: {
      slug: 'RS',
      name: 'Rio Grande do Sul',
    },
    absolute_url: '/brasil/rs/gramado/',
  },
  {
    id: '2',
    image: '/images/mocks/places/place-2.jpg',
    city: {
      slug: 'florianopolis',
      name: 'Florian칩polis',
    },
    country: {
      slug: 'brasil',
      name: 'Brasil',
    },
    state: {
      slug: 'SC',
      name: 'Santa Catarina',
    },
    absolute_url: '/brasil/sc/florianopolis/',
  },
  {
    id: '3',
    image: '/images/mocks/places/place-3.jpg',
    city: {
      slug: 'curitiba',
      name: 'Curitiba',
    },
    country: {
      slug: 'brasil',
      name: 'Brasil',
    },
    state: {
      slug: 'PR',
      name: 'Paran치',
    },
    absolute_url: '/brasil/pr/curitiba/',
  },
  {
    id: '4',
    image: '/images/mocks/places/place-4.jpg',
    city: {
      slug: 'rio-de-janeiro',
      name: 'Rio de Janeiro',
    },
    country: {
      slug: 'brasil',
      name: 'Brasil',
    },
    state: {
      slug: 'RJ',
      name: 'Rio de Janeiro',
    },
    absolute_url: '/brasil/rj/rio-de-janeiro/',
  },
  {
    id: '5',
    image: '/images/mocks/places/place-5.jpg',
    city: {
      slug: 'maceio',
      name: 'Macei칩',
    },
    country: {
      slug: 'brasil',
      name: 'Brasil',
    },
    state: {
      slug: 'AL',
      name: 'Alagoas',
    },
    absolute_url: '/brasil/al/maceio/',
  },
  {
    id: '6',
    image: '/images/mocks/places/place-6.jpg',
    city: {
      slug: 'natal',
      name: 'Natal',
    },
    country: {
      slug: 'brasil',
      name: 'Brasil',
    },
    state: {
      slug: 'RN',
      name: 'Rio Grande do Norte',
    },
    absolute_url: '/brasil/rn/natal/',
  },
  {
    id: '7',
    image: '/images/mocks/places/place-7.jpg',
    city: {
      slug: 'fortaleza',
      name: 'Fortaleza',
    },
    country: {
      slug: 'brasil',
      name: 'Brasil',
    },
    state: {
      slug: 'CE',
      name: 'Cear치',
    },
    absolute_url: '/brasil/ce/fortaleza/',
  },
  {
    id: '8',
    image: '/images/mocks/places/place-8.jpg',
    city: {
      slug: 'porto-seguro',
      name: 'Porto Seguro',
    },
    country: {
      slug: 'brasil',
      name: 'Brasil',
    },
    state: {
      slug: 'BA',
      name: 'Bahia',
    },
    absolute_url: '/brasil/ba/porto-seguro/',
  },
];
