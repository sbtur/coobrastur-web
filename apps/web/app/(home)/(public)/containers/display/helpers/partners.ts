export const PARTNERS = [
  {
    alt: 'Coob+',
    src: '/images/shared/logos/coob.png',
    width: 112,
    height: 25,
  },
  {
    alt: 'Viajar Faz Bem',
    src: '/images/shared/logos/viajar-faz-bem.png',
    width: 101,
    height: 27,
  },
  {
    alt: 'Meridien',
    src: '/images/shared/logos/meridien.png',
    width: 120,
    height: 35,
  },
  {
    alt: 'Basntur',
    src: '/images/shared/logos/banstur.png',
    width: 125,
    height: 23,
  },
];
