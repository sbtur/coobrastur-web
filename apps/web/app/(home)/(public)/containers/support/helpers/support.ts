import { Support } from '../types/support.types';

export const SUPPORT: Support[] = [
  {
    icon: '/images/icons/clara.png',
    title: 'Dúvidas? Fale com a Clara',
    description:
      'Conte com o atendimento exclusivo da Clara, nossa especialista virtual, para uma experiência personalizada e eficiente.',
    href: 'https://wa.me/5511999999999',
    target: '_blank',
  },
  {
    icon: '/images/icons/phone-call.svg',
    title: 'Atendimento Exclusivo',
    description:
      'Nossos agentes são treinados e estão prontos para ouvir suas necessidades, oferecendo soluções rápidas e eficazes.',
    href: 'https://wa.me/5511999999999',
  },
  {
    icon: '/images/icons/luggage.svg',
    title: 'Agência de Viagens',
    description:
      'Oferecemos uma agência de viagens com soluções personalizadas para tornar sua experiência ainda mais incrível.',
    href: 'https://wa.me/5511999999999',
  },
];
