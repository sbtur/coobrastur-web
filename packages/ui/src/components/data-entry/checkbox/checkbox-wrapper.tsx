export const CheckboxWrapper = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  return (
    <div className="flex flex-row items-start space-x-3 space-y-0">
      {children}
    </div>
  );
};
