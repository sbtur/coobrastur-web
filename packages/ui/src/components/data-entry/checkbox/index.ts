export * from './checkbox';
export * from './checkbox-wrapper';
