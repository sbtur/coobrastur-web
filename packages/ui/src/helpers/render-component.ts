import { ComponentType, createElement } from 'react';

import { render } from '@testing-library/react';

export const renderComponent = <Props extends Record<string, unknown>>(
  Component: ComponentType<Props>,
  props: Props,
) => {
  return render(createElement(Component, props));
};
