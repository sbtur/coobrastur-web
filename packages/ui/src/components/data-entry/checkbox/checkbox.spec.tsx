import { Checkbox } from './checkbox';

import { renderComponent } from '@ui/helpers/render-component';
import { beforeEach, describe, expect, it, userEvent, vi } from '@ui/tests';

describe('Checkbox', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('should match snapshot', () => {
    expect(renderComponent(Checkbox, {})).toMatchSnapshot();
  });

  it('should render unchecked by default', () => {
    const { getByTestId } = renderComponent(Checkbox, {
      id: 'unchecked-checkbox',
      'data-testid': 'unchecked-checkbox',
    });

    expect(getByTestId('unchecked-checkbox')).not.toBeChecked();
  });

  it('should render as checked when checked prop is true', () => {
    const { getByTestId } = renderComponent(Checkbox, {
      id: 'checked-checkbox',
      checked: true,
      'data-testid': 'checked-checkbox',
    });

    expect(getByTestId('checked-checkbox')).toBeChecked();
  });

  it('should call onCheckedChange when clicked', async () => {
    const onCheckedChange = vi.fn();
    const { getByTestId } = renderComponent(Checkbox, {
      id: 'on-checked-change-checkbox',
      onCheckedChange,
      'data-testid': 'on-checked-change-checkbox',
    });

    await userEvent.click(getByTestId('on-checked-change-checkbox'));
    expect(onCheckedChange).toHaveBeenCalledTimes(1);
    expect(onCheckedChange).toHaveBeenCalledWith(true);
  });

  it('should be disabled when disabled prop is true', () => {
    const { getByTestId } = renderComponent(Checkbox, {
      id: 'disabled-checkbox',
      disabled: true,
      'data-testid': 'disabled-checkbox',
    });

    expect(getByTestId('disabled-checkbox')).toBeDisabled();
  });

  it('should not trigger onCheckedChange when disabled', async () => {
    const onCheckedChange = vi.fn();
    const { getByTestId } = renderComponent(Checkbox, {
      id: 'disabled-on-checked-change-checkbox',
      disabled: true,
      onCheckedChange,
      'data-testid': 'disabled-on-checked-change-checkbox',
    });

    await userEvent.click(getByTestId('disabled-on-checked-change-checkbox'));
    expect(onCheckedChange).not.toHaveBeenCalled();
  });

  it('should apply custom className', () => {
    const { getByTestId } = renderComponent(Checkbox, {
      id: 'custom-class-checkbox',
      className: 'custom-class',
      'data-testid': 'custom-class-checkbox',
    });

    expect(getByTestId('custom-class-checkbox')).toHaveClass('custom-class');
  });
});
