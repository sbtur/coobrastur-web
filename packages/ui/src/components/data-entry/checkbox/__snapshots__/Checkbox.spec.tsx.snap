// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-checked="false"
        class="peer h-5 w-5 shrink-0 rounded-md border border-input shadow-sm focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-blue-500 data-[state=checked]:border-input data-[state=checked]:text-white"
        data-state="unchecked"
        role="checkbox"
        type="button"
        value="on"
      />
    </div>
  </body>,
  "container": <div>
    <button
      aria-checked="false"
      class="peer h-5 w-5 shrink-0 rounded-md border border-input shadow-sm focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-blue-500 data-[state=checked]:border-input data-[state=checked]:text-white"
      data-state="unchecked"
      role="checkbox"
      type="button"
      value="on"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
